### ISSUE TITLE
Handle 503 proxy target missing error

### ISSUE THREAD

**Issue by carolynvs** [carolynvs/howtowhale#24]
When JupyterHub can't connect to the proxy target, i.e. the user's docker container, the 503 error message sucks. What I would really like to do instead is handle that error, treat the old server as stopped and restart it.

**Scenario 1: Delete the cluster**
1. Log in and start your server.
2. Open a notebook.
3. From the Carina website or command line client, uncerimoniously delete the cluster upon which your user's Jupyter server is running.
4. Refresh the notebook page. In this case I get a 503 error. There is a sweet spot where I can end up either getting a 504 timeout (because the cluster is rebuilding) or I end up on the status page that tells my server is slowly starting.

**Scenario 2: Delete the container**
1. Log in and start your server.
2. Open a notebook.
3. From the command line, load your cluster's credentials then use docker to delete the Jupyter server.
4. Refresh the notebook page. In this case, I get a 503 error. But after waiting a minute, when I refresh my server has been rebuilt.

Basically when we hit a 503 error due to the user's server being unexpectedly gone (i.e. still registered in the JupyterHub proxy), I want to automatically restart their server, retriggering a `spawner.stop` then `spawner.start`, so that I can cleanly recover without showing ugly messages to the user.

For what it's worth, in other cases JupyterHub handles this pretty well. If I am on the JupyterHub control panel when I delete the Docker container running the Jupyter, the polling process eventually realizes that the server is gone and rebuilds it.
